app/models/sub_service_request.rb

```diff
@@ -72,6 +72,15 @@ class SubServiceRequest < ApplicationRecord
   after_save :update_org_tree
   after_save :update_past_status
 
+  # Users should not be able to delete a request that has been pushed
+  # to Fulfillment
+  before_destroy do
```

Comment:
@6reg This is looking good so far! I pulled down your branch and did some testing and I don't think this is functioning as intended. First thing I noticed is that while the ssr is not being deleted (if `in_work_fulfillment? == true`) it's still allowing the last service to be deleted, and all the emails are still going out. Also I'm not seeing any kind of error message or anything. I think an alternate solution would be to get rid of the `before_destroy` callback, have a conditional in the view that disabled the delete button if it was the last and pushed to fulfillment, and if that's the case display a tooltip on why it can't be deleted. After reading the story I noticed another fun twist to this. If any of the services under an ssr that has been pushed to fulfillment have been invoiced (there's an invoiced flag somewhere, it probably lives in fulfillment and you'll have to use the api to get to it), they also can't be deleted.

Reply:
Let me know if you want to pair on this!

Reply:
Btw, it looks like invoiced is at the fulfillment level in cwf.

Reply:
@6reg Also, the way you would access that would be `Shard::Fulfillment::Fulfillment` or something like that. You might have to play around with it. Look through `app/models/shard` for more examples and how they relate to our actual sparc models.